Whiskey/Tasks/AppVeyorWaitForBuildJobs.ps1

```diff
@@ -0,0 +1,99 @@
+
+function Wait-WhiskeyAppVeyorBuildJob
+{
+    [CmdletBinding()]
+    [Whiskey.Task('AppVeyorWaitForBuildJobs')]
+    param(
+        [Parameter(Mandatory)]
+        [Whiskey.Context]$TaskContext,
+
+        [Parameter(Mandatory)]
+        [string]$ApiKeyID,
+
+        [TimeSpan]$CheckInterval = '00:00:10',
+
+        [TimeSpan]$ReportInterval = '00:01:00',
+
+        [string[]]$InProgressStatus = @('running','queued'),
+
+        [string[]]$SuccessStatus = @('success')
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState

```

Comment:
Maybe add an `$ErrorActionPreference = 'Stop'` to this task, so that if anything unexpected happens we fail the build rather than the task short-circuiting continuing with the rest of the publishing tasks.

Reply:
🤔

Reply:
Any task could be used in the Publish portion of a pipeline. Shouldn't we have `$ErrorActionPreference = 'Stop'` in every task for the same reason?

Reply:
Yes, I agree we should have that for all tasks and I actually thought we already did in `Invoke-WhiskeyTask` but turns out we don't. We mostly rely on calling `Stop-WhiskeyTask` to throw an exception during the build. I wasn't sure if there was a historical reason or not why we aren't using `$ErrorActionPreference = 'Stop'` for all tasks.

Reply:
Because it's PowerShell and not all errors are errors worth stopping for.